diff --speed-large-files --minimal -Nru tmp/gcc-3.4.3/libstdc++-v3/docs/doxygen/user.cfg.in work/gcc-3.4.3/libstdc++-v3/docs/doxygen/user.cfg.in
--- tmp/gcc-4.2.2/libstdc++-v3/docs/doxygen/user.cfg.in	2004-05-22 06:07:19.000000000 +0100
+++ work/gcc-4.2.2/libstdc++-v3/docs/doxygen/user.cfg.in	2005-03-26 01:36:53.030462000 +0000
@@ -619,14 +619,14 @@
 # Microsoft HTML help workshop to generate a compressed HTML help file (.chm) 
 # of the generated HTML documentation.
 
-GENERATE_HTMLHELP      = NO
+GENERATE_HTMLHELP      = YES
 
 # If the GENERATE_HTMLHELP tag is set to YES, the CHM_FILE tag can 
 # be used to specify the file name of the resulting .chm file. You 
 # can add a path in front of the file if the result should not be 
 # written to the html output directory.
 
-CHM_FILE               = 
+CHM_FILE               = libstdc++.chm
 
 # If the GENERATE_HTMLHELP tag is set to YES, the HHC_LOCATION tag can 
 # be used to specify the location (absolute path including file name) of 
